internal/terminal/table.go

```diff
@@ -0,0 +1,149 @@
+package terminal
+
+import (
+	"fmt"
+	"github.com/fatih/color"
+	"strings"
+)
+
+// table
+// headers is an ordered list of strings to specify the order of printing
+// data is a map of a string to another string to print, where the key should be a value in headers
+//	if not, we will not print
+// maxHeaderWidth is to determine the column width based on the max length of the values and header
+type table struct {
+	headers        []string
+	data           []map[string]string
+	maxHeaderWidth map[string]int
+}
+
+// columnBreakWidth is to set a standard for the break between columns for the table
+const (
+	 columnBreakWidth = 2
+)
+
+func newTable(headers []string, values []map[string]interface{}) *table {
```

Comment:
I return a pointer so that I could return a nil instead of a table if the values are empty (header, data) 
I also have a question in regards to the other comment that involves implementing a validate function, since if I return a nil here, there should be no real way for them to create a table without going through the initial validation

Reply:
yeah let's avoid using a pointer just to return `nil` here, the zero-value of the struct will be just fine